### ISSUE TITLE
Keras vs Tensorflow (loss is very different)

### ISSUE THREAD

**Issue by zeeshanahmad19** [aymericdamien/TensorFlow-Examples#265]
I have a made a model for binary classification of diabetes using pima-indian-diabetes dataset. Code is running in both cases but loss fluctuations in case of tensorflow are very high. I have tried using same learning rate but results are very different. I'm not getting the problem. **Am I doing it right?**
Need some help!!!
@aymericdamien

**Tensorflow Code:**
```
import tensorflow as tf
import pandas as pd
import numpy as np
from math import ceil

data = pd.read_csv('pima-indians-diabetes.csv', header=None)
data.head()

X_train = data.values[..., 0:8]
Y_train = data.values[..., 8]

Y_train = Y_train.reshape(-1, 1)
Y_train.shape

# Parameters
learning_rate = 0.0001
epochs = 150
batch_size = 10

# Network Parameters
n_hidden_1 = 12
n_hidden_2 = 8 
num_input = 8 
num_classes = 1

# tf Graph input
X = tf.placeholder(X_train.dtype, [None, X_train.shape[1]])
Y = tf.placeholder(Y_train.dtype, [None, Y_train.shape[1]])

# Store layers weight & bias
weights = {
    'h1': tf.Variable(tf.random_normal([num_input, n_hidden_1], dtype=tf.float64)),
    'h2': tf.Variable(tf.random_normal([n_hidden_1, n_hidden_2], dtype=tf.float64)),
    'out': tf.Variable(tf.random_normal([n_hidden_2, num_classes], dtype=tf.float64))
}
biases = {
    'b1': tf.Variable(tf.random_normal([n_hidden_1], dtype=tf.float64)),
    'b2': tf.Variable(tf.random_normal([n_hidden_2], dtype=tf.float64)),
    'out': tf.Variable(tf.random_normal([num_classes], dtype=tf.float64))
}

# Create model
def neural_net(x):
    # Hidden fully connected layer with 256 neurons
    layer_1 = tf.add(tf.matmul(x, weights['h1']), biases['b1'])
    layer_1 = tf.nn.relu(layer_1)
    # Hidden fully connected layer with 256 neurons
    layer_2 = tf.add(tf.matmul(layer_1, weights['h2']), biases['b2'])
    layer_2 = tf.nn.relu(layer_2)
    # Output fully connected layer with a neuron for each class
    out_layer = tf.matmul(layer_2, weights['out']) + biases['out']
    return out_layer

# Construct model
logits = neural_net(X)

softmax_result = tf.nn.sigmoid_cross_entropy_with_logits(
    logits=logits, labels=Y)
# Define loss and optimizer
loss_op = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(
    logits=logits, labels=Y))
optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate)
train_op = optimizer.minimize(loss_op)

# Initialize the variables (i.e. assign their default value)
init = tf.global_variables_initializer()

#Function using to get batches of data...
def get_batch(features, target, batch_size=20, random=False):
    data = np.hstack((np.asarray(features), np.asarray(target).reshape(-1, 1)))
    if random is True:
        np.random.shuffle(data)
    for i in range(0, len(data)):
        if i%batch_size == 0:
            yield {'features':data[i:i+batch_size, 0:-1], 'target': data[i:i+batch_size, -1:]}


with tf.Session() as sess:

    # Run the initializer
    sess.run(init)
    for i in range(10):
        j = 1
        for batch_data in get_batch(X_train, Y_train, 10, True):
            #print("batch ", j)
            #print(batch_data['features'], batch_data['target'])
            sess.run(train_op, feed_dict={X: batch_data['features'], Y: batch_data['target']})
            loss = sess.run(loss_op, feed_dict={X: batch_data['features'], Y: batch_data['target']})
            j+=1
        
        print("loss: ", loss)
        
                
    
                
    

```

**Tensorflow Output**

```
loss:  47.0994501493
loss:  91.8339402843
loss:  46.1353108953
loss:  96.7579063661
loss:  71.2980093301
loss:  73.1184740308
loss:  31.0596809836
loss:  47.9874363086
loss:  1.47742377622
loss:  9.32253628485

After Epoch 10 loss : 9.32253628485
```

**Keras Code:**
```
from keras.models import Sequential
from keras.layers import Dense
import numpy

dataset = numpy.loadtxt("pima-indians-diabetes.csv", delimiter=",")

X = dataset[:,0:8]
Y = dataset[:,8]

model = Sequential()
model.add(Dense(12, input_dim=8, activation='relu'))
model.add(Dense(8, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

model.fit(X, Y, epochs=150, batch_size=10)

scores = model.evaluate(X, Y)
print("\n%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))
```
**Keras Output:**
```
Epoch 1/10
768/768 [==============================] - 0s 621us/step - loss: 4.3369 - acc: 0.6563
Epoch 2/10
768/768 [==============================] - 0s 134us/step - loss: 3.5715 - acc: 0.6406
Epoch 3/10
768/768 [==============================] - 0s 146us/step - loss: 1.9766 - acc: 0.6393
Epoch 4/10
768/768 [==============================] - 0s 131us/step - loss: 1.6612 - acc: 0.6458
Epoch 5/10
768/768 [==============================] - 0s 155us/step - loss: 1.3763 - acc: 0.6432
Epoch 6/10
768/768 [==============================] - 0s 190us/step - loss: 1.1545 - acc: 0.6484
Epoch 7/10
768/768 [==============================] - 0s 199us/step - loss: 0.9848 - acc: 0.6497
Epoch 8/10
768/768 [==============================] - 0s 208us/step - loss: 0.8263 - acc: 0.6732
Epoch 9/10
768/768 [==============================] - 0s 187us/step - loss: 0.7678 - acc: 0.6654
Epoch 10/10
768/768 [==============================] - 0s 159us/step - loss: 0.7571 - acc: 0.6706
768/768 [==============================] - 0s 107us/step

acc: 71.09%
After Epoch 10 loss : 0.7571

```
**Highlight the issue if I'm doing something wrong!!!**
